control/tests/test_heal.py

```diff
@@ -165,6 +165,62 @@ def test_determine_fault(self, edx_id, video_trans_status, video_trans_start, vi
             for e in encode_list:
                 self.assertTrue(e in self.encode_list)
 
+    @data(
+        {
+            'edx_id': '1',
+            'video_trans_status': 'Corrupt File',
+            'video_trans_start': datetime.datetime.utcnow().replace(tzinfo=utc),
+            'video_active': True,
+        },
+        {
+            'edx_id': '1',
+            'video_trans_status': 'Review Reject',
+            'video_trans_start': datetime.datetime.utcnow().replace(tzinfo=utc),
+            'video_active': True,
+        },
+        {
+            'edx_id': '1',
+            'video_trans_status': 'Review Hold',
+            'video_trans_start': datetime.datetime.utcnow().replace(tzinfo=utc),
+            'video_active': True,
+        },
+        {
+            'edx_id': '2',
+            'video_trans_status': 'Complete',
+            'video_trans_start': datetime.datetime.utcnow().replace(tzinfo=utc),
+            'video_active': False,
+        },
+        {
+            'edx_id': '2',
+            'video_trans_status': 'Ingest',
+            'video_trans_start': datetime.datetime.utcnow().replace(tzinfo=utc),
+            'video_active': True,
+        },
+        {
+            'edx_id': '1',
+            'video_trans_status': 'Corrupt File',
+            'video_trans_start': datetime.datetime.utcnow().replace(tzinfo=utc),
+            'video_active': True,
+        },
+    )
+    @unpack
+    def test_determine_fault_freeze_bug(self, edx_id, video_trans_status, video_trans_start, video_active):
+        video_instance = Video(
+            edx_id=edx_id,
+            video_trans_status=video_trans_status,
+            video_trans_start=video_trans_start,
+            video_active=video_active,
+            inst_class=self.course_object
+        )
+        video_instance.save()
+        heal_instance_two = VedaHeal(freezing_bug=True)
+        encode_list = heal_instance_two.determine_fault(video_instance)
+        if video_instance.edx_id == '1':
```

Comment:
@yro we should avoid branch on `ddt` while asserting the expectations. 

As long as we pass the expectations along with the `ddt`, we wouldn't be falling in a situation where we need to branch on `ddt` for asserting the expectations. 

May you please address this. Think about just having `self.assertItemsEqual(actual_encode_list, ddt_passed_expectations)` instead of `the if/else`s.

Reply:
@yro Please see the below [test](https://github.com/edx/edx-video-pipeline/blob/master/VEDA/tests/test_utils.py#L71-L110) to see how we should write ddt in unit tests. Let us know if you need any help. 

```python
    @data(
        {
            'urls': ('http://api.cielo24/', '/add/job'),
            'params': {},
            'expected_url': 'http://api.cielo24/add/job'
        },
        {
            'urls': ('http://api.cielo24', '/add/job'),
            'params': {'a': 1, 'b': 2},
            'expected_url': 'http://api.cielo24/add/job?a=1&b=2'
        },
        {
            'urls': ('http://api.cielo24/', 'add/job'),
            'params': {'c': 3, 'd': 4},
            'expected_url': 'http://api.cielo24/add/job?c=3&d=4'
        },
        {
            'urls': ('http://api.cielo24', 'add/job'),
            'params': {'p': 100},
            'expected_url': 'http://api.cielo24/add/job?p=100'
        },
        {
            'urls': ('http://api.cielo24', 'add/job', 'media'),
            'params': {'p': 100},
            'expected_url': 'http://api.cielo24/add/job/media?p=100'
        }
    )
    @unpack
    def test_build_url(self, urls, params, expected_url):
        """
        Tests that utils.build_url works as expected.
        """
        url = utils.build_url(
            *urls,
            **params
        )
        self.assertEqual(
            url,
            expected_url
        )
```

Reply:
@Qubad786 OK.